app/controllers/hotels_controller.rb

```diff
@@ -1,14 +1,68 @@
 class HotelsController < ApplicationController
-  before_action :set_hotel, only: [:show]
+  before_action :set_hotel, only: %i[show edit update destroy]
+  before_action :set_labels, only: %i[new edit]
+
   def index
-    @hotels = Hotel.all
-    @hotels_count = Hotel.count
+    @hotels = policy_scope(Hotel).all
+    @hotels_count = policy_scope(Hotel).count
```

Comment:
Why do you create @hotel_count? You can call @hotel.count. Also try to use scope from the line above without all